providers/implementations/kdfs/pbkdf2.c

```diff
@@ -154,7 +158,11 @@ static void kdf_pbkdf2_init(KDF_PBKDF2 *ctx)
         /* This is an error, but there is no way to indicate such directly */
         ossl_prov_digest_reset(&ctx->digest);
     ctx->iter = PKCS5_DEFAULT_ITER;
-    ctx->lower_bound_checks = ossl_kdf_pbkdf2_default_checks;
+#ifdef FIPS_MODULE
+    ctx->lower_bound_checks = FIPS_pbkdf2_lower_bound_check(provctx);
```

Comment:
This doesnt seem right... it should just be set to 1 here..
The config option should be read during the on_unapproved.

Reply:
It just uses the settable, and unless the settable is set to 0 (it is -1 to start with) it will be in strict mode.

Reply:
Fixed. The commit for this change has been dropped.